flapflapio/simulator
# FLAP-47, FLAP-48: testing and refactoring

# Description

This is a pretty beefy PR which adds:

1. Unit tests for most of the packages in our app
2. Some nice refactoring and clean up to the core packages
3. Many bug fixes throughout the app

## Link to work item

Closes FLAP-47
Closes FLAP-48

## How to Test

Check out the refactored packages inside `core`. I made a lot of nice tweaks to
simplify our domain model.

To run the new testing suite:

```bash
go clean --testcache
go test ./...
```
